File: 1-1.py
example = """1abc2
pqr3stu8vwx
a1b2c3d4e5f
treb7uchet"""


def solve(input):
    digits = (
        [c for c in line if c.isdigit()] for line in input.split("\n") if len(line) > 0
    )
    numbers = (int(ds[0] + ds[-1]) for ds in digits)
    return sum(numbers)


if __name__ == "__main__":
    assert solve(example) == 142
    with open("1.txt", "r") as file:
        print(solve(file.read()))

File: 1-1.c
#include "prelude.h"

static const char *example =
	"1abc2\n"
	"pqr3stu8vwx\n"
	"a1b2c3d4e5f\n"
	"treb7uchet\n";

static int solve(const char *input) {
	int sum = 0;
	while (*input) {
		char digits[3] = {0};
		for (uint i = 0; *input && *input != '\n'; input++) {
			if (!isdigit(*input)) continue;
			if (!i) digits[i++] = *input;
			digits[i] = *input;
		}
		sum += atoi(digits);
		input++;
	}
	return sum;
}

int main(void) {
	const int example_result = solve(example);
	if (example_result != 142) {
		printf("FAIL: expected 142 actual %d\n", example_result);
		return 1;
	}
	printf("Result: %d\n", solve(readToString("1.txt")));
	return 0;
}
